crates/solver-types/src/standards/eip7683.rs

```diff
@@ -415,46 +415,51 @@ impl TryFrom<&Quote> for interfaces::StandardOrder {
 impl interfaces::StandardOrder {
 	/// Handle Permit2 order conversion
 	fn handle_permit2_quote_conversion(
-		_quote: &Quote,
+		quote: &Quote,
 		eip712_data: &serde_json::Map<String, serde_json::Value>,
 	) -> Result<Self, Box<dyn std::error::Error>> {
+		tracing::info!("Starting permit2 quote conversion with new format");
 		use crate::utils::parse_bytes32_from_hex;
 		use alloy_primitives::{Address, U256};
 		use interfaces::SolMandateOutput;
 
-		// Extract signing information to get domain details
-		let signing = eip712_data
-			.get("signing")
-			.and_then(|s| s.as_object())
-			.ok_or("Missing 'signing' object in EIP-712 data")?;
-
-		let domain = signing
-			.get("domain")
-			.and_then(|d| d.as_object())
-			.ok_or("Missing 'domain' in signing data")?;
-
-		// Extract origin chain ID from domain
-		let origin_chain_id = domain
-			.get("chainId")
-			.and_then(|c| c.as_u64())
-			.map(U256::from)
-			.ok_or("Missing chainId in domain")?;
+		// Extract origin chain ID from the quote's payload domain (new format)
+		let origin_chain_id = if let crate::OifOrder::OifEscrowV0 { payload } = &quote.order {
+			// Get domain from payload directly (new format)
+			let domain = payload
+				.domain
+				.as_object()
+				.ok_or("Missing 'domain' object in payload")?;
+
+			domain
+				.get("chainId")
+				.and_then(|c| c.as_str().and_then(|s| s.parse::<u64>().ok()))
+				.map(U256::from)
+				.ok_or("Missing chainId in domain")?
+		} else {
+			return Err("Expected OifEscrowV0 order type".into());
+		};
```

Comment:
This seems like it should be a helper fn `extract_chain_ids(OifOrder)` which contains logic for either type (escrow, 3009, resource lock)